AsyncUsageAnalyzers/AsyncUsageAnalyzers/Naming/AvoidAsyncSuffixCodeFixProvider.cs

```diff
@@ -0,0 +1,54 @@
+﻿namespace AsyncUsageAnalyzers.Naming
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Threading.Tasks;
+    using Helpers;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+
+    /// <summary>
+    /// Implements a code fix for <see cref="AvoidAsyncSuffixAnalyzer"/>.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = nameof(AvoidAsyncSuffixCodeFixProvider))]
+    [Shared]
+    public class AvoidAsyncSuffixCodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(AvoidAsyncSuffixAnalyzer.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                if (!diagnostic.Id.Equals(AvoidAsyncSuffixAnalyzer.DiagnosticId))
+                {
+                    continue;
+                }
+
+                var token = root.FindToken(diagnostic.Location.SourceSpan.Start);
+                if (token.IsMissing)
+                {
+                    continue;
+                }
```

Comment:
It seems quite unlikely (probably impossible) that you would have a missing token here.